presto-main/src/test/java/com/facebook/presto/operator/TestExchangeClient.java

```diff
@@ -329,6 +341,100 @@ public void testClose()
         assertStatus(clientStatus, location, "closed", "not scheduled");
     }
 
+    @Test
+    public void testInitialRequestLimit()
```

Comment:
Please make sure you run this in a loop for at least 100 - 1000 times, to make sure there's no flakines in the test